boards/Makefile.common

```diff
@@ -176,7 +176,7 @@ all: release
 # binary. This makes checking for Rust errors much faster.
 .PHONY: check
 check:
-	$(Q)RUSTFLAGS="$(RUSTFLAGS_FOR_CARGO)" $(CARGO) check $(VERBOSE) --target=$(TARGET) --package $(PLATFORM) --release
+	$(Q)$(CARGO) check $(VERBOSE)
```

Comment:
What are the consequences of removing these flags?
- `--package`: Shouldn't this be scoped to the current board (when running `make check` within a board), instead of the whole workspace?
- `--target`: Some code (e.g. in arch) depends on the target CPU. Is this checking the intended code?

Reply:
re `--package`, [according to the docs](https://doc.rust-lang.org/stable/cargo/reference/workspaces.html#the-workspace-section):
> If neither of those flags are specified, Cargo will use the package in the current working directory.

Empirically, this does this thing (i.e. only checks `hail` & dependencies in the hail folder)

---

re `--target`, I mentioned this in the commit message: yes, it does cause check to skip over any `cfg(arch`-gated code. However, there is very little of this in Tock, and the rules that actually build targets will catch them. The gain is that crates are not recompiled when you change which board you're checking for (as no flags to cargo have changed). This is a significant speedup on my machine